azurerm/resource_arm_cosmos_db_account.go

```diff
@@ -390,28 +726,62 @@ func flattenAndSetAzureRmCosmosDBAccountFailoverPolicy(d *schema.ResourceData, l
 	d.Set("failover_policy", &results)
 }
 
-func resourceAzureRMCosmosDBAccountConsistencyPolicyHash(v interface{}) int {
+func flattenAndSetAzureRmCosmosDBAccountGeoLocations(d *schema.ResourceData, account documentdb.DatabaseAccount) error {
+	locationSet := schema.Set{
+		F: resourceAzureRMCosmosDBAccountGeoLocationHash,
+	}
+
+	//we need to propagate the `prefix` field so fetch existing
+	prefixMap := map[string]string{}
+	for _, lRaw := range d.Get("geo_location").(*schema.Set).List() {
+		lb := lRaw.(map[string]interface{})
+		prefixMap[lb["location"].(string)] = lb["prefix"].(string)
+	}
+
+	for _, l := range *account.FailoverPolicies {
+		id := *l.ID
+		lb := map[string]interface{}{
+			"id":                id,
+			"location":          azureRMNormalizeLocation(*l.LocationName),
+			"failover_priority": int(*l.FailoverPriority),
+		}
+
+		//if id is not the default then it must be set via prefix
+		if id != resourceArmCosmosDBAccountGenerateDefaultId(d.Get("name").(string), lb["location"].(string)) {
+			lb["prefix"] = id
+		}
+
+		locationSet.Add(lb)
+	}
+
+	return d.Set("geo_location", &locationSet)
+}
+
+//todo remove once deprecated field `failover_policy` is removed
+func resourceAzureRMCosmosDBAccountFailoverPolicyHash(v interface{}) int {
 	var buf bytes.Buffer
 	m := v.(map[string]interface{})
 
-	consistencyLevel := m["consistency_level"].(string)
-	maxInterval := m["max_interval_in_seconds"].(int)
-	maxStalenessPrefix := m["max_staleness_prefix"].(int)
+	location := azureRMNormalizeLocation(m["location"].(string))
+	priority := int32(m["priority"].(int))
 
-	buf.WriteString(fmt.Sprintf("%s-%d-%d", consistencyLevel, maxInterval, maxStalenessPrefix))
+	buf.WriteString(fmt.Sprintf("-%s-%d", location, priority))
```

Comment:
if we're changing the hash value I think we'll need a state migration here?

Reply:
Good spot, the `-` got in there by accident.